OBC: added idle and init files

Riaabz opened the pull request:
I added idle and init files

Riaabz committed:
OBC: added idle and init files

I added idle and init files

Signed-off-by: Riaab Zahid <riaab.zahid@gmail.com>

alexapostolu commented on apps/obc/src/obc_check.c:
```diff
@@ -0,0 +1,56 @@
+    return ping_ok && flash_ok && RTC_ok;
+}
+
+
```
Only need one space at the end of the file

angelazheng96 replied:
Fixed in #138

Riaabz committed:
Idle: Fixed space at end

Fixed space at end

Signed-off-by: Riaab Zahid <riaab.zahid@gmail.com>

mshinjo reviewed (commented):
Good start! Let's use snake_case, and please address the CI failure.

mshinjo commented:
Also, let's make use of `static` unless the functions need to be publicly exposed APIs.

Riaabz committed:
Idle: changed idle.c and idle.h
I reviewed the comments on the idle.c and idle.h files and corrected them. I also changed the logic of idle.c a bit

Signed-off-by: Riaab Zahid <riaab.zahid@gmail.com>

Riaabz committed:
Init commit

Init commit

Signed-off-by: Riaab Zahid <riaab.zahid@gmail.com>

angelazheng96 commented on apps/obc/src/idle.c:
```diff
@@ -0,0 +1,125 @@
+ * @startuml //need plantuml support to be able to generate diagram from mermaid code comments 
+ *  
+ * 
+*/
```
Resolving - outdated
